### PR TITLE
P2-215 Create Select instruction.

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* [schema-blocks] Creates a Select block instruction to the schema-blocks package. With this instruction, a `select` element can be added to a schema template. 

## Relevant technical choices:

* Decided to make this into a single `select` element, so multiple values cannot be selected currently.
	* The issues and designs of the schema templates currently do not call for `select`s that expect multiple values.
	* I would propose, if we need one, to add it as a separate instruction when needed.
		* There are fundamental differences in how the values are stored (multiple instead of one) and in the design of the `select` element. 
* The `createElement` import is now ignored for the `no-unused-vars` ESLint warning.
	* `createElement` is needed to create React elements, but is never called directly when using JSX. This leads to a false positive in ESLint. 		
* I adapted the transform configuration in the Jest configuration so Jest is able to parse `tsx` files. 	 

## Test instructions

This PR can be tested by following these steps:

* In WordPress SEO, in a schema template (for example `src/schema-templates/recipe.block.php`), add a `Select` instruction.
	* E.g. like this:
	```
	{{select label="Cuisine" name="cuisine" options={ "Caribbean": "caribbean", "German": "german", "Cantonese": "cantonese" } }}
	``` 
* Build the plugin as you normally would.
	* Make sure that your monorepo is linked and on the right branch. 
	* Make sure that you have built the `schema-blocks` package using `yarn build`. 	
	* Make sure that you run `grunt build:dev` in the plugin, so the changes from the `schema-blocks` are integrated within the JavaScript files of the plugin.  	
* Create a new post.
* Add a recipe block.
* See that it contains your added `select` instruction in the form of a drop down box with a label.
* Select an option in the drop down box.
* Save the post.
* View the post on the frontend of your website.
* See that it shows the label and selected value in the recipe.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * N/A: this PR only affects the `schema-blocks` package, which is not yet released and behind a feature flag.

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

### CODE REVIEW

File: packages/schema-blocks/src/instructions/blocks/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import BlockInstruction from "../../core/blocks/BlockInstruction";
+import { RenderEditProps, RenderSaveProps } from "../../core/blocks/BlockDefinition";
+import { createElement, ReactElement } from "@wordpress/element";
+import { BlockConfiguration, BlockInstance } from "@wordpress/blocks";
+import attributeExists from "../../functions/validators/attributeExists";
+import attributeNotEmpty from "../../functions/validators/attributeNotEmpty";
```
Comment (andizer) [inline packages/schema-blocks/src/instructions/blocks/Select.tsx:6]: You might consider introducing an `validators/index.ts` that exposes the validators.

**Review by andizer — CHANGES_REQUESTED** [review]
CR done

File: packages/schema-blocks/src/instructions/blocks/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	 *
+	 * @param props The render props.
+	 *
+	 * @returns The label of the selected option.
+	 */
+	protected label( props: RenderSaveProps | RenderEditProps ): string {
+		if ( Array.isArray( this.options.options ) ) {
+			// Value is the same as the label, so just return the value.
+			return this.value( props );
+		}
```
[the remark this answers is not in the document]
Comment (hansjovis) [reply packages/schema-blocks/src/instructions/blocks/Select.tsx:67]: The instruction can get the options that can be selected in two ways. One: as an array of possible options. E.g. `[ "option 1", "option 2", "option 3" ]`. Two: as an object mapping labels to values `{ "label-1": "value 1", "label-2": "value 2" }`. It was in the issue description and it is similarly implemented in the sidebar select instruction.

In the first case, the values and the labels are the same. E.g. `"option 1"` is both the label and the value.

I will make the comment more clear.

**Review by andizer — APPROVED** [review]
CR 👍

**Comment by johannadevos** [comment]
Acceptance: ✅

**Pull request merged**